File: leetcode/src/Num538.java
public class Num538 {
    class Solution {
        /**
         * RVL中序遍历会得到一个逆序的序列，遍历过程中记录下来遍历过的点的值的和
         * */
        private int sum = 0;
        public TreeNode convertBST(TreeNode root) {
            if(root == null){return null;}
            // 先遍历右子树
            convertBST(root.right);
            // 当前节点加上前面的数的和
            root.val+=sum;
            // 更新sum
            sum = root.val;
            // 再遍历左节点
            convertBST(root.left);
            return root;
        }
    }
}

File: learnPython/num538.py
# -*- coding: utf-8 -*-
"""
Created on Thu Oct 25 20:44:33 2018

@author: Fay
"""

class Solution(object):
    sum = 0
    def convertBST(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        if root == None:
            return root
        self.convertBST(root.right)
        self.sum = self.sum + root.val
        root.val = self.sum
        self.convertBST(root.left)
        return root